src/main/flight/pid.c

```diff
@@ -524,17 +550,40 @@ static void detectAndSetCrashRecovery(
     }
 }
 
-static void handleItermRotation()
+static void rotateVector(float v[3], float rotation[3]) 
 {
-    // rotate old I to the new coordinate system
-    const float gyroToAngle = dT * RAD;
-    for (int i = FD_ROLL; i <= FD_YAW; i++) {
+    // rotate v around rotation vector rotation
+    // rotation in radians, all elements must be small
+    for (int i = 0; i < 3; i++) {
         int i_1 = (i + 1) % 3;
         int i_2 = (i + 2) % 3;
-        float angle = gyro.gyroADCf[i] * gyroToAngle;
-        float newPID_I_i_1 = pidData[i_1].I + pidData[i_2].I * angle;
-        pidData[i_2].I -= pidData[i_1].I * angle;
-        pidData[i_1].I = newPID_I_i_1;
+        float newV = v[i_1] + v[i_2] * rotation[i];
+        v[i_2] -= v[i_1] * rotation[i];
+        v[i_1] = newV;
+    }
+}
+
+static void handleRotations() 
+{
+    if (itermRotation || acGain > 0) {
+        const float gyroToAngle = dT * RAD;
+        float rotationRads[3];
+        for (int i = FD_ROLL; i <= FD_YAW; i++) {
+            rotationRads[i] = gyro.gyroADCf[i] * gyroToAngle;
+        }
+        if (acGain > 0) {
+            rotateVector(axisError, rotationRads);
```

Comment:
This is quite hidden update

Reply:
Do you have a better suggestion? You don't like the in/out parameter?

Reply:
It isn't obvious that `handleRotations` modifies axisError and pidData. axisError may be passed as argument, pidData[].I is  harder ... 

Maybe rotateITermAndAxisError? I hate long names, but `handleRotations();` will surely bite someone in future ...

Reply:
Actually I just incrementally changed the previous source which called handleItermRotation. But yeah I guess that removed a bit of info in the name. I'll add you suggestion.